packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -273,6 +275,11 @@ class ToolbarOptions {
   ///
   /// Defaults to false. Must not be null.
   final bool selectAll;
+
+  /// The actions that appear in the toolbar.
+  ///
+  /// Defaults to empty. Must not be null.
```

Comment:
This should explain that empty actions means you just get the default options: cut, copy, paste, and select all.  Anything added to `actions` results in all of the defaults plus your action after the defaults.

Which begs the question, is this the behavior we want?  Currently, this implementation does not support the following features:

  1. The developer can't change the order of the default actions, or put custom actions before or in between the default actions. 
  1. The developer can't remove the default actions.
  1. The developer can't change the behavior of the default actions.

Maybe it's a good thing to not support these features. Are any of these features doable on native?

We could do something like default `actions` to an array containing cut, copy, paste, and select all, but that way might get complicated when handling default behavior and when actions are hidden.

Reply:
Also, could you add an example in the docs?  Maybe here is a good place.  Use the `/// {@tool snippet}` syntax (no need to use a dartpad example, because dartpad on the web uses the browser's context menu).  Something simple like what's in the test is fine.

Reply:
2. removing the default ones is as simple as passing false to the `ToolBarOptions` constructor.

1. So one work around for reordering is to just disable the default ones and add custom versions. None are super complicated. This is how one would achieve the same result on native android. 

3. Same solution as above

There are good arguments for not changing the default order, except it might be nice to insert an action before the default ones, and the current solution doesn't support that with rolling your own versions of the defaults.

One solution would be to move the logic for the standard operations out of `EditableTextState` and into constructors `TextSelectionAction.copy()`,  `TextSelectionAction.paste()`, etc. I would have done that except that the cupertino controls and material controls use divergent localization.

Reply:
Thanks, I think you're right that all three of these are actually possible, which is reassuring.  It's probably fine to not include `TextSelectionAction.copy()` etc. in this PR.  We can wait and see if there is demand for making these kinds of features easier.